ugs-core/src/com/willwinder/universalgcodesender/uielements/helpers/SteppedSizeManager.java

```diff
@@ -0,0 +1,79 @@
+/*
+    Copyright 2016-2017 Will Winder
+
+    This file is part of Universal Gcode Sender (UGS).
+
+    UGS is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    UGS is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with UGS.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.willwinder.universalgcodesender.uielements.helpers;
+
+import javax.swing.JComponent;
+import java.awt.Dimension;
+import java.awt.event.ComponentAdapter;
+import java.awt.event.ComponentEvent;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Track size of a component and fire event when nominal size changes.
+ */
+public class SteppedSizeManager {
```

Comment:
Intrigued by this. Could the sizes be used for things like buttons and images as well? Or is it mainly for labels?

Reply:
Because of the apparent need to pre-generate fonts at specific sizes, it's meant for labels. (That's why it's "stepped" rather than "continuous".) But sure, it can be used for anything that can be resized.

Reply:
Marking as "nothing to do" for tracking my tasks.